codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -794,48 +702,58 @@ protected void initialWindowSize(int newWindowSize) throws Http2Exception {
         }
 
         @Override
-        public void enqueueFrame(AbstractState state, FlowControlled frame) throws Http2Exception {
+        void enqueueFrame(FlowState state, FlowControlled frame) throws Http2Exception {
             super.enqueueFrame(state, frame);
             checkConnectionThenStreamWritabilityChanged(state);
         }
 
         @Override
-        public void stateCancelled(AbstractState state) {
+        void stateCancelled(FlowState state) {
             try {
                 checkConnectionThenStreamWritabilityChanged(state);
             } catch (Http2Exception e) {
-                logger.error("Caught unexpected exception from checkAllWritabilityChanged", e);
+                throw new RuntimeException("Caught unexpected exception from checkAllWritabilityChanged", e);
             }
         }
 
         @Override
-        public void channelWritabilityChange() throws Http2Exception {
-            if (connectionState.markWritability() != isChannelWritable()) {
+        void channelWritabilityChange() throws Http2Exception {
+            if (connectionState.markedWritability() != isChannelWritable()) {
                 checkAllWritabilityChanged();
             }
         }
 
-        private void notifyWritabilityChanged(AbstractState state) {
-            state.markWritability(!state.markWritability());
+        private void checkStateWritability(FlowState state) throws Http2Exception {
+            if (isWritable(state) != state.markedWritability()) {
+                if (state == connectionState) {
+                    checkAllWritabilityChanged();
+                } else {
+                    notifyWritabilityChanged(state);
+                }
+            }
+        }
+
+        private void notifyWritabilityChanged(FlowState state) {
+            state.markedWritability(!state.markedWritability());
             try {
                 listener.writabilityChanged(state.stream);
-            } catch (RuntimeException e) {
-                logger.error("Caught unexpected exception from listener.writabilityChanged", e);
+            } catch (Throwable cause) {
+                logger.error("Caught Throwable from listener.writabilityChanged", cause);
             }
         }
 
-        private void checkConnectionThenStreamWritabilityChanged(AbstractState state) throws Http2Exception {
+        private void checkConnectionThenStreamWritabilityChanged(FlowState state) throws Http2Exception {
             // It is possible that the connection window and/or the individual stream writability could change.
-            if (isWritableConnection() != connectionState.markWritability()) {
+            if (isWritableConnection() != connectionState.markedWritability()) {
                 checkAllWritabilityChanged();
-            } else if (isWritable(state) != state.markWritability()) {
+            } else if (isWritable(state) != state.markedWritability()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Prefix this call to "isWritable" with "super.". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2388)